lxd/certificates.go

```diff
@@ -449,10 +479,8 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 				return response.Forbidden(fmt.Errorf("No matching certificate add operation found"))
 			}
 
-			tokenReq, ok := joinOp.Metadata["request"].(api.CertificatesPost)
-			if !ok {
-				return response.InternalError(fmt.Errorf("Bad certificate add operation data"))
-			}
+			// We have already validated that the request metadata is present and ensured it is this type in certificateTokenValid.
```

Comment:
Can we just return a api.CertificatesPost from certificateTokenValid rather than an operation?

Reply:
Ah yeah sorry I missed that part

Reply:
Done.